### PR TITLE
docs: Reorganize Storybook sidebar nav to better reflect USWDS docs

### LINKED ISSUE #784: [docs] Reorganize Storybook component navigation
**Does this documentation relate to a specific USWDS feature?**
No

**Is your documentation related to a problem or inconsistency? Please describe.**
Most of the components in this library map directly to a USWDS component, but some of them are composed of multiple components, or are Truss-designed components (and should maybe be moved into a different library).

**Describe the documentation you'd like**
Update the Storybook sidebar navigation so that it is as easy as possible to find USWDS components (consider following the same structure as https://designsystem.digital.gov/components/), and organize Truss & other components separately.

### PR DESCRIPTION
**Pull request by suzubara**
# Summary

In the interest of making components easier to find and match to their USWDS counterparts, this PR organizes components in Storybook under groups:
- **Design tokens:** stories that illustrate or exemplify USWDS [design tokens](https://designsystem.digital.gov/design-tokens/), [utility classes](https://designsystem.digital.gov/utilities/), or other plain HTML that is not part of an exported React component
- **Components:** stories for exported USWDS [components](https://designsystem.digital.gov/components/) and sub-components
- **Page templates:** stories that illustrate USWDS [page templates](https://designsystem.digital.gov/page-templates/)
- **Other:** everything else, such as Truss-designed components. This group may not be long for this world.

## Related Issues or PRs

Closes #784 

## How To Test

- Start Storybook (`yarn storybook`)
- Review the ordering and grouping of sidebar navigation 

### Screenshots (optional)

image

### CODE REVIEW

**Comment by trussworks-infra** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>This PR does not include changes to tests, even though it affects source code.</td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against e7607d4c4563b11d38713b6ccc3edbeea1ee7c5f
</p>

File: src/components/ButtonGroup/ButtonGroup.stories.tsx [hunk 1]
```diff
@@ -4,7 +4,7 @@ import { Button } from '../Button/Button'
 import { Link } from '../Link/Link'
 
 export default {
-  title: 'ButtonGroup',
+  title: 'Components/Button groups',
```
Comment (brandonlenz) [inline src/components/ButtonGroup/ButtonGroup.stories.tsx:7]: Minor question: I noticed most of the time `title` follows the convention `Components/ComponentName`, but a few (like this one) use `Components/Component name`.

Is it worth aligning that across all stories?

Comment (suzubara) [reply src/components/ButtonGroup/ButtonGroup.stories.tsx:7]: @brandonlenz good call, I tried to match the navigation used in the sidebar on https://designsystem.digital.gov/components/

My thinking was that someone using this lib is probably coming from USWDS and wants to be able to easily find our components from that perspective (and I figure they can find the name of the component itself using the docs or source code)

Comment (brandonlenz) [reply src/components/ButtonGroup/ButtonGroup.stories.tsx:7]: Got it. That makes sense to me then!

**Review by brandonlenz — DISMISSED** [review]

File: .storybook/preview.ts [hunk 2]
```diff
@@ -4,11 +4,16 @@ import 'uswds/dist/css/uswds.css'
...
-import { addParameters } from '@storybook/client-api'
 import { INITIAL_VIEWPORTS } from '@storybook/addon-viewport'
 
 export const parameters = {
   viewport: {
     viewports: INITIAL_VIEWPORTS,
   },
+  options: {
+    storySort: {
+      method: 'alphabetical',
```
Comment (haworku) [inline .storybook/preview.ts:15]: nice

**Review by brandonlenz — APPROVED** [review]
Looks great! I'll make sure to update the stories for my Breadcrumb PR with this as well.

**Review by haworku — APPROVED** [review]

**Pull request merged**